### PR TITLE
More aggressive retry for connection_closed and debug logging

### PR DESCRIPTION
**Pull request by forest**

### CODE REVIEW

File: lib/snowpack.ex [hunk 1]
```diff
@@ -152,8 +154,10 @@ defmodule Snowpack do
   @spec query(conn, iodata, list, [query_option()]) ::
           {:ok, Snowpack.Result.t()} | {:error, Exception.t()}
   def query(conn, statement, params \\ [], options \\ []) when is_iodata(statement) do
-    # retry after 50 ms only once and do not rescue exceptions
-    retry with: [50], rescue_only: [] do
+    # retry after 50 ms 3 times
+    retry with: [50] |> Stream.cycle() |> Stream.take(3) do
```
Comment (forest) [inline lib/snowpack.ex:158]: Now also rescuing Runtime exceptions and retrying.

File: test/snowpack/protocol_test.exs [hunk 2]
```diff
@@ -31,7 +31,15 @@ defmodule Snowpack.ProtocolTest do
     end
 
     @tag :capture_log
-    test "with connection_closed error query is retried ONLY once", %{pid: pid} do
+    test "with connection_closed error query is retried 3 times", %{pid: pid} do
+      expect Snowpack.ODBC, :query, fn _pid, _statement, _params, _opts, _with_query_id ->
+        {:error, Snowpack.Error.exception(:connection_closed)}
+      end
```
Comment (aschreifels) [inline test/snowpack/protocol_test.exs:37]: I think if we do this we can eliminate the extra expects. :)

```suggestion
      expect Snowpack.ODBC, :query, 3, fn _pid, _statement, _params, _opts, _with_query_id ->
        {:error, Snowpack.Error.exception(:connection_closed)}
      end
```

**Review by aschreifels — CHANGES_REQUESTED** [review]

File: test/snowpack/protocol_test.exs (hunk 2 above)
Comment (forest) [reply test/snowpack/protocol_test.exs]: Forgot about that. :)

**Review by pdgonzalez872 — APPROVED** [review]

**Pull request merged**